packages/firestore/src/remote/remote_store.ts

```diff
@@ -248,7 +230,7 @@ export class RemoteStore {
    * from watch so we wait for the ack to process any messages from this target.
    */
   private sendWatchRequest(queryData: QueryData): void {
-    this.recordPendingTargetRequest(queryData.targetId);
+    this.watchChangeAggregator.recordPendingTargetRequest(queryData.targetId);
```

Comment:
The watch change aggregator wouldn't have to ask if a target is active if it kept track of it itself.

For each `WatchTargetChange.Added` event: add to targetStates
For each `WatchTargetChange.Removed` event: remove from targetStates

One difference is that `WatchTargetChange.Reset` can't remove from targetStates but actually has to clear the entry.

Reply:
This unfortunately doesn't quite work since a target is only active if the View is still interested in it _and_ if it has no pending ACKs. A target is usually long inactive before the Watch stream finishes sending us the REMOVE messages.

I think we can simplify this a lot if Views ignored updates for inactive targets, but I would rather not make this change right now.